README.md

```diff
@@ -13,3 +13,139 @@ This is an NWB extension for storing metadata of devices used in optical experim
+    }
+    class PulsedExcitationSource{
+        <<DeviceModel>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        peak_power_in_W : numeric, optional
+        peak_pulse_energy_in_J : numeric, optional
+        pulse_rate_in_Hz : numeric, optional
+    }
+    class Photodetector{
+        <<DeviceModel>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        detector_type : text, optional
+        detected_wavelength_in_nm : numeric, optional
+        gain : numeric, optional
+    }
+    class DichroicMirror{
+        <<DeviceModel>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        cut_on_wavelength_in_nm : numeric, optional
+        cut_off_wavelength_in_nm : numeric, optional
+        reflection_bandwidth_in_nm : numeric, optional
+        transmission_bandwidth_in_nm : numeric, optional
+        angle_of_incidence_in_degrees : numeric, optional
+    }
+    class OpticalFilter{
+        <<DeviceModel>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        filter_type : text, optional
+    }
+    class BandOpticalFilter{
+        <<OpticalFilter>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        center_wavelength_in_nm : numeric
+        bandwidth_in_nm : numeric
+    }
+    class EdgeOpticalFilter{
+        <<OpticalFilter>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        cut_wavelength_in_nm : numeric
+        slope_in_percent_cut_wavelength : numeric, optional
+        slope_starting_transmission_in_percent : numeric, optional
+        slope_ending_transmission_in_percent : numeric, optional
+    }
+    class ObjectiveLens{
+        <<DeviceModel>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        numerical_aperture : numeric, optional
+        magnification : numeric, optional
+    }
+    class Microscope{
+        <<DeviceModel>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        microscopy_type : text, optional
+        doi : text, optional
+    }
+    OpticalFiber *-- DeviceModel : extends
+    ExcitationSource *-- DeviceModel : extends
+    PulsedExcitationSource *-- ExcitationSource : extends
+    Photodetector *-- DeviceModel : extends
+    DichroicMirror *-- DeviceModel : extends
+    Microscope *-- DeviceModel : extends
+    ObjectiveLens *-- DeviceModel : extends
+    OpticalFilter *-- DeviceModel : extends
+    BandOpticalFilter *-- OpticalFilter : extends
+    EdgeOpticalFilter *-- OpticalFilter : extends
```

Comment:
Overall this looks great, thanks!

The other thing I was suggesting was instead of having one massive diagram you could separate each 'theme' or 'group' and repeat the `DeviceModel` definition in each

Like, one diagram for the optical filters; one for the excitation source, another for fibre photometry specific stuff, another for the indicator/effort (which should not inherit from `Device` and should therefore have `manufacturer` specified on them)